scipy/interpolate/_aaa.py

```diff
@@ -0,0 +1,404 @@
+    rational functions on a real or complex set of points. The rational approximation is
+    represented in a barycentric form from which the roots (zeros), poles, and residues
+    can be computed.
+
+    Parameters
+    ----------
+    z : 1D array_like
+        Values at which `f` is provided.
+    f : 1D array_like
+        Function values ``f(z)`` at `z`. Infinite and NaN value of `f` and corresponding
+        values of `z` will be discarded.
+    rtol : float, optional
+        Relative tolerance, defaults to ``eps**0.75``.
+    max_terms : int, optional
+        Maximum number of terms in the barycentric representation, defaults to 100.
+
+    Attributes
+    ----------
+    support_points : array
+        Support points of the approximation.
+    values : array
+        Value of the approximation at the `support_points`.
+    weights : array
+        Weights of the barycentric approximation.
+    errors : array
+        Error in the successive iterations of AAA.
+    
+    Methods
+    -------
+    poles()
+        Compute the poles of the rational approximation.
+    residues()
+        Compute the residues associated with the `poles` of the rational approximation.
+    roots()
+        Compute the roots(zeros) of the rational approximation.
+    
+
+    Warns
+    -----
+    RuntimeWarning
+        If `rtol` is not achieved in `max_terms` iterations.
+
+    See Also
+    --------
+    pade : Padé approximation
+
+    Notes
+    -----
+    At the :math:`m` th iteration (at which point there are :math:`m` terms), the
+    rational approximation in the AAA algorithm takes the barycentric form
+
+    .. math::
+
+        r(z) = n(z)/d(z) =
+        \frac{\sum_{j=1}^m\ w_j f_j / (z - z_j)}{\sum_{j=1}^m w_j / (z - z_j)},
+
+    where :math:`z_1,\dots,z_m` are real or complex support points selected from `z`,
+    :math:`f_1,\dots,f_m` are a set of real or complex data values, and
+    :math:`w_1,\dots,w_m` are real or complex weights. The algorithm then proceeds to
+    select the next support point :math:`z_{m+1}` is selected from the remaining
+    unselected points in `z` such that the nonlinear residual :math:`|f(z) - n(z)/d(z)|`
+    is maximised. The algorithm terminates when this maximum is less than
+    ``rtol * np.linalg.norm(f, ord=np.inf)``. This means the interpolation property is
+    only satisfied up to a tolerance. The weights are selected to solve the
+    least-squares problem
+
+    .. math::
+
+        \text{minimise}|fd - n| \quad \text{subject to} \quad
+        \sum_{i=1}^{m+1} w_i = 1,
+
+    over the unselected points in `z`.
+
+    References
+    ----------
+    .. [1] Y. Nakatsukasa, O. Sete, and L. N. Trefethen, "The AAA algorithm for
+            rational approximation", SIAM J. Sci. Comp. 40 (2018), A1494-A1522.
+            :doi:`10.1137/16M110612`
+
+    Examples
+    --------
+
+    Here we reproduce a number of the numerical examples from [1]_ as demonstration
```

Comment:
Consider "as *a* demonstration of the functionality" or "to demonstrate the functionality".